proxy/service_util.go

```diff
@@ -0,0 +1,38 @@
+package proxy
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+func (s *workflowServiceProxyServer) reqCtx(ctx context.Context) context.Context {
+	if s.disableHeaderForwarding {
+		return ctx
+	}
+
+	// Copy incoming header to outgoing if not already present in outgoing. We
+	// have confirmed in gRPC that incoming is a copy so we can mutate it.
+	incoming, _ := metadata.FromIncomingContext(ctx)
```

Comment:
Error check?

Reply:
This isn't an error, this is whether it was present, and we don't care really in our case